### DISCUSSION TITLE
How to use client-side load balancing with Vector as source

### DISCUSSION THREAD

**Discussion by hannuriha** [vectordotdev/vector#16449] [Q&A] [answered]
Hi

In the documentation below, it says that vector is capable of client-side load balancing.
https://vector.dev/docs/setup/going-to-prod/architecting/#aggregator-role
![image](https://user-images.githubusercontent.com/3232220/218984738-d51c89fb-b721-4065-8fdb-0b45a9b6d9fb.png)

However, it is difficult to find the client-side load balancing settings on https://vector.dev/docs/reference/configuration/sinks/vector/.

I would like to know which ones are client load balancing in vector.

Please let me know.
Regards,

---
EDIT:

ChatGPT said the wrong answer
![image](https://user-images.githubusercontent.com/3232220/218992305-8eb66f99-bb44-49b3-8ef5-62bf69175a07.png)

**Comment by jszwedko** [comment, marked as answer]
Hi @hannuriha !

You are correct, that diagram is inaccurate. Only the `elasticsearch` sink currently supports client-side load balancing. We had considered extending this to the `vector` sink which is probably why the diagram has it there. For now, load balancing requires a deployed load balancer like nginx or haproxy.

Neat use of ChatGPT but it does seem to be giving an incorrect answer.
